godotproject/Menus/InGameMenu/InGameMenu.gd

```diff
@@ -91,14 +91,15 @@ func _ready():
 func _process(_delta):
 	var menu_pressed = Input.is_action_just_pressed("menu")
 	# We use whether this node is visible to track whether the menu is open.
-	if menu_pressed && screen_current == null:
+	var inhibit_pause = "inhibit_pause" in get_tree().current_scene && get_tree().current_scene.inhibit_pause
+	if !inhibit_pause && menu_pressed && screen_current == null:
 		open_menu()
 	elif visible:
 		global_main()
 	
 	# Apply our variable pause_child - it will be animated during screen fades
 	if screen_current != null:
-		screen_current.pause_mode = PAUSE_MODE_STOP if pause_child else PAUSE_MODE_PROCESS
+		screen_current.pause_mode = PAUSE_MODE_STOP if pause_child else PAUSE_MODE_INHERIT
```

Comment:
We want to avoid setting "PAUSE_MODE_PROCESS" wherever possible so that the scope of "what's running" vs. "what's not" is clear. If this is set to PROCESS then the user can still interact with the menu even after a global fade starts which is not intended.